website_form_recaptcha/models/website_form_recaptcha.py

```diff
@@ -80,32 +82,56 @@ def validate_response(self, response, remote_ip, website=None):
 
         if not res.get('success'):
             raise ValidationError(self._get_error_message())
-
         return True
 
     @api.model
     def validate_request(self, request, values):
-        # Only check once: if a call to reCAPTCHA's API is made twice with
-        # the same token, we get a 'timeout-or-duplicate' error. So we
-        # stick to the first response data storing the token after the
-        # first invoke in the current request object. This duplicated
-        # call can be cause for instance by website_crm_phone_validation.
-        try:
-            getattr(request, self.RESPONSE_ATTR)
-        except AttributeError:
-            return None
+        old_value = getattr(request, self.REQUEST_TOKEN, None)
+        if old_value is not None:
+            # Only check once: if a call to reCAPTCHA's API is made twice with
+            # the same token, we get a 'timeout-or-duplicate' error. So we
+            # stick to the first response data storing the token after the
+            # first invoke in the current request object. This duplicated
+            # call can be cause for instance by website_crm_phone_validation.
+            return True
+        req_value = values.get(self.RESPONSE_ATTR)
+        if not req_value:
+            raise ValidationError(
+                self._get_error_message('missing-input-secret')
+            )
         ip_addr = request.httprequest.environ.get('HTTP_X_FORWARDED_FOR')
         if ip_addr:
             ip_addr = ip_addr.split(',')[0]
         else:
             ip_addr = request.httprequest.remote_addr
-        try:
-            self.validate_response(
-                values.get(self.RESPONSE_ATTR), ip_addr
-            )
-            # Store reCAPTCHA's token in the current request object
-            setattr(
-                request, self.RESPONSE_ATTR, values.get(self.RESPONSE_ATTR)
-            )
-        except ValidationError:
-            raise ValidationError([self.RESPONSE_ATTR])
+        # if not validated an exception is raised anyway
+        validated = self.validate_response(req_value, ip_addr)
+        # Store reCAPTCHA's token in the current request object
+        setattr(request, self.REQUEST_TOKEN, req_value)
+        return validated
+
+    # TODO: backward compat, remove in v12
+    @api.model
+    def action_validate(self, req_value, ip_addr):
+        """Backward compatibility for old implementation.
+
+        Prior to API refactoring pre-validation steps
```

Comment:
Could you specify which API refactoring? Maybe put a link or something. I'm confused if it's Odoo's, OCA's, or Google's.